docs/modules/architecture/pages/vector-search-overview.adoc

```diff
@@ -0,0 +1,122 @@
+
+== Partitioned similarity search
+
+Vector indexes are partitioned, so when you execute similarity search all partitions need to be searched and partial results aggregated.
+It is important to understand the impact of that process on search performance and recall.
+
+=== Two-stage search
+
+The default search algorithm is two-stage search which works as follows:
+
+1. Member that received the query is a coordinator.
+2. Coordinator fans out search request to each member (including self). Each member is tasked with returning results from partitions it owns.
+3. Each member executes search on owned partitions in parallel and aggregates the partition results.
+4. Partially aggregated results are returned to coordinator.
+5. Coordinator aggregates partial results and generates final results.
+   If necessary, searches on some partitions can be retried individually, for example in case of migrations, members leaving the cluster or other errors.
+
+Aggregation at each stage is done based on score and only the best results are kept.
+
+There are 2 important parameters in this search algorithm that determine amount of data sent between the members and quality of the final result:
+
+- `partitionLimit` - number of search results obtained from each partition
+- `memberLimit` - number of search results returned from member to coordinator
+
+If `topK` denotes number of entries requested, then the following should hold in order for the system to be able to return enough results:
+
+- `partitionLimit * partitionCount >= topK`, `partitionLimit &lt;= topK`
+- `memberLimit * memberCount >= topK`, `memberLimit &lt;= topK`
+
+By default `partitionLimit` and `memberLimit` are equal to `topK` which satisfies the above inequalities but causes processing of much more results than the requested `topK`.
+This improves the overall quality of the results but can have a significant performance overhead because more entries are fetched from each partition of the index and sent between the members.
+
+NOTE: Consider tuning `partitionLimit` based on quality and latency requirements. It should be adjusted alongside number of partitions.
+`memberLimit` is less critical for overall behavior if there are only a few members.
+
+=== Single-stage search
+
+A simplified search algorithm exists which does not perform intermediate aggregation of results on member level.
+It is used if the cluster has only a single member. It can be also enabled using search hint in other cases.
+
+In this case search request is executed in parallel on all partitions (on their owners)
+and partition results are aggregated directly on the coordinator member to produce the final result.
+
+This search algorithm has `partitionLimit` parameter which behaves in the same way as for two-stage search.
+
+== Partition count impact
+
+Number of partitions has big impact on the performance of the vector collection. There are numerous conflicting factors impacting selection of optimal partition count:
```

Comment:
```suggestion
The number of partitions has a big impact on the performance of the vector collection. Other conflicting factors that can impact the selection of an optimal partition count including the following:
```

Reply:
those are not "other factors", these are _the_ factors :)

Reply:
Note amended, @k-jamroz WDYT?